bagger/src/bagger.py

```diff
@@ -147,23 +153,37 @@ def load_xml(path):
     return load_from_string(xml_string)
 
 
-def dispatch_bag(bag_details):
-    # now zip this bag in a way that will be efficient for the archiver
-    logging.debug("creating zip file for " + bag_details["b_number"])
+def pack(bag_details):
+    archive_format = settings.ARCHIVE_FORMAT
+
+    logging.debug(" ".join(["packing", archive_format, bag_details["b_number"]]))
 
-    shutil.make_archive(
-        bag_details["zip_file_path"][0:-4], "zip", bag_details["directory"]
+    base_name = "/".join([settings.WORKING_DIRECTORY, bag_details["b_number"]])
+
+    zip_file_path = shutil.make_archive(
+        base_name, archive_format, bag_details["directory"]
     )
 
-    logging.debug("uploading " + bag_details["zip_file_name"] + " to S3")
+    zip_file_name = zip_file_path.split("/")[-1]
+
+    bag_details["zip_file_path"] = zip_file_path
+    bag_details["zip_file_name"] = zip_file_name
+
+    return bag_details
+
 
+def dispatch(bag_details):
     upload_location = aws.upload(
         bag_details["zip_file_path"], bag_details["zip_file_name"]
     )
 
-    logging.debug("upload completed")
+    logging.debug(
+        " ".join(["uploaded", bag_details["zip_file_name"], f"{upload_location}"])
```

Comment:
ditto above, I'd prefer
```suggestion
        "uploaded %s %s", bag_details["zip_file_name"], upload_location)
```

also, `f"{upload_location}"` seems redundant?